docker/Dockerfile

```diff
@@ -12,43 +12,267 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM ubuntu:focal AS base
+# Create basic runtime environment base on Ubuntu 22.04 (jammy)
+# Define build arguments
+# Create and activate runtime virtual environment
+FROM ubuntu:jammy AS base-runtime
+ARG KOKKOS_VERSION=4.1.00
+ARG LIGHTNING_VERSION=0.32.0
+ARG CUDA_VERSION=12.2
+ARG CUDA_ARCH=AMPERE80
+ARG AMD_ARCH=AMD_GFX90A
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+    apt-utils \
+    ca-certificates \
+    libgomp1 \
+    python3 \
+    python3-pip \
+    python3-venv \
+    tzdata \
+    wget \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+ENV VIRTUAL_ENV=/opt/venv
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
-# Setup and install basic packages
+# Create basic build environment with build tools and compilers
+FROM base-runtime AS base-build
 RUN apt-get update \
-    && apt-get install --no-install-recommends -y apt-utils \
-    && DEBIAN_FRONTEND="noninteractive" \
-    apt-get install --no-install-recommends -y tzdata \
+    && apt-get install --no-install-recommends -y \
     build-essential \
-    ca-certificates \
     ccache \
     cmake \
     curl \
     git \
-    python3 \
-    python3-pip \
-    python3-venv \
+    ninja-build \
     python3-dev \
     gcc-11 g++-11 cpp-11 \
     && apt-get clean \
-    && rm -rf /var/lib/apt/lists/* \
-    && update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 100 --slave /usr/bin/g++ g++ /usr/bin/g++-11 --slave /usr/bin/gcov gcov /usr/bin/gcov-10 \
-    && /usr/sbin/update-ccache-symlinks \
-    && mkdir /opt/ccache \
-    && ccache --set-config=cache_dir=/opt/ccache \
-    && python3 -m venv /opt/venv
+    && rm -rf /var/lib/apt/lists/*
+RUN update-alternatives \
+    --install /usr/bin/gcc gcc /usr/bin/gcc-11 100 \
+    --slave /usr/bin/g++ g++ /usr/bin/g++-11 \
+    --slave /usr/bin/gcov gcov /usr/bin/gcov-10
+RUN /usr/sbin/update-ccache-symlinks
+RUN mkdir /opt/ccache
+RUN ccache --set-config=cache_dir=/opt/ccache
```

Comment:
Is it worth combining these run commands to help reduce the base-build image size? Or is that no longer required?